### PR TITLE
Support an optional namespace for looking up globals

### PR DESCRIPTION
**Pull request by cbiesinger**
Needed for GDB.

### CODE REVIEW

File: extensions/jsdbg/core/jsdbg.js [hunk 1]
```diff
@@ -446,14 +446,18 @@ Loader.OnLoad(function () {
...
                 {name:"symbol", type:"string", description:"The symbol to evaluate."},
                 {name:"typeName", type:"string", description: "The type name of the symbol to look up."},
+                {name:"namespace", type:"string", description: "The namespace the symbol is in. Required on some platforms."},
                 {name:"callback", type:"function(object)", description:"A callback that is called when the operation succeeds or fails."}
             ]
         },
-        LookupGlobalSymbol: function(module, symbol, typeName, callback) {
+        LookupGlobalSymbol: function(module, symbol, typeName, namespace, callback) {
+            var namespaceStr = "";
+            if (namespace)
```
Comment (sanketj) [inline extensions/jsdbg/core/jsdbg.js]: Consider applying the same pattern to typename. This feels cleaner than having multiple calls to JsonRequest.

File: server/JsDbg.Core/IDebugger.cs [hunk 2]
```diff
@@ -112,7 +112,7 @@ public interface IDebugger {
         Task<IEnumerable<SConstantResult>> LookupConstants(string module, string type, ulong constantValue);
         Task<SConstantResult> LookupConstant(string module, string type, string constantName);
         Task<SFieldResult> LookupField(string module, string typename, string fieldName);
-        Task<SSymbolResult> LookupGlobalSymbol(string module, string symbol, string typeName);
+        Task<SSymbolResult> LookupGlobalSymbol(string module, string symbol, string typeName, string nameSpace);
```
Comment (sanketj) [inline server/JsDbg.Core/IDebugger.cs]: nit: name**s**pace, to be consistent with the client

File: server/JsDbg.Core/WebServer.cs [hunk 3]
```diff
@@ -21,7 +21,7 @@
 using System.IO;
 
 namespace JsDbg.Core {
-    
+
```
Comment (sanketj) [inline server/JsDbg.Core/WebServer.cs]: Are these newlines deliberate?

File: server/JsDbg.Windows/DiaDebugger.cs [hunk 4]
```diff
@@ -299,7 +299,9 @@ public class DiaDebugger : IDebugger {
             return (await this.LoadType(module, typename)).Size;
         }
 
-        public async Task<SSymbolResult> LookupGlobalSymbol(string moduleName, string symbolName, string typeName) {
+        public async Task<SSymbolResult> LookupGlobalSymbol(string moduleName, string symbolName, string typeName, string nameSpace) {
```
Comment (sanketj) [inline server/JsDbg.Windows/DiaDebugger.cs]: nit: same here, lower-case 's'

File: extensions/jsdbg/core/jsdbg.js (hunk 1 above)
Comment (cbiesinger) [reply extensions/jsdbg/core/jsdbg.js]: Done

File: server/JsDbg.Core/IDebugger.cs (hunk 2 above)
Comment (cbiesinger) [reply server/JsDbg.Core/IDebugger.cs]: Unfortunately namespace is a reserved identifier and I can't use it as a parameter name :(

File: server/JsDbg.Core/WebServer.cs (hunk 3 above)
Comment (cbiesinger) [reply server/JsDbg.Core/WebServer.cs]: My editor "helpfully" removed trailing whitespace, let me see if I can find a good way to fix that...

File: server/JsDbg.Core/IDebugger.cs (hunk 2 above)
Comment (sanketj) [reply server/JsDbg.Core/IDebugger.cs]: Oh yeah, sorry I missed that.

**Comment by cbiesinger** [comment]
OK, newline changes removed, this should be ready to merge

**Pull request merged**